Update swift versions

jandamm opened the pull request:
This will use a more modern version of Swift and also a newer version of Publish.

With these changes the plugin should work just fine for any newly created Publish project.

jandamm committed:
Modernize Swift and dependencies

jandamm committed:
Update Package.swift

labradon commented:
Thanks for your efforts in updating the package. Could you help me out with the motivation of changing the Publish version compatibility? As far as I see it, specifying the Publish dependency as `.package(url: "https://github.com/johnsundell/publish.git", from: "0.8.0")` restricts the usage. The package is compatible with all versions of Publish since "0.1.0", so I have my doubts whether this change is a good way to go forward. Do I misunderstand something here or do you have other motivations for this change?

A second addition: According to [https://www.timc.dev/posts/understanding-swift-packages/](https://www.timc.dev/posts/understanding-swift-packages/), if the package name, url and target name match, the dependency for the target can just be specified as a string as already done `"Publish"`. Looking into the Publish package specification, it is named as `"Publish"`. Only the URL doesn't match at the moment. We could fix that leading to the following `Package.swift`:

```swift
// swift-tools-version:5.5
import PackageDescription

let package = Package(
    name: "MinifyCSSPublishPlugin",
    platforms: [.macOS(.v12)],
    products: [
        .library(
            name: "MinifyCSSPublishPlugin",
            targets: ["MinifyCSSPublishPlugin"]),
    ],
    dependencies: [
        .package(url: "https://github.com/johnsundell/Publish.git", from: "0.1.0"),
    ],
    targets: [
        .target(
            name: "MinifyCSSPublishPlugin",
            dependencies: ["Publish"]),
        .testTarget(
            name: "MinifyCSSPublishPluginTests",
            dependencies: ["MinifyCSSPublishPlugin"]),
    ]
)
```

Let me know what you think. If you are okay with these changes and include them into your merge request, I'll do the merge ASAP.

jandamm commented:
Yeah so I was trying to use the plugin in a newly setup Publish website.
This didn't work because `Publish` needs macOS 12 or later and your plugin didn't require any platform so it defaults to `10.13` which resulted in an error.

Then I added the `v12` requirement and thought users who use macOS 12 would probably use Swift 5.5 anyhow.
Using Swift 5.5 resulted in `dependency 'Publish' in target 'MinifyCSSPublishPlugin' requires explicit declaration; reference the package in the target dependency with '.product(name: "Publish", package: "publish")'`. So I changed `"Publish"` to `.product(name: "Publish", package: "publish")`.

Bumping Publish to 0.8.0 wasn't for any particular reason apart from users using an older version would probably still get the previous release. And others would get the new version.

I can try what is the least amount of changes necessary to get things building again. I applied those fixes on the GitHub web interface and just pulled it in via SPM.

jandamm committed:
Reduce minimum requirement of Publish to 0.1.0

jandamm commented:
Just checked it again.

All changes except the minimum version of Publish were necessary in order to include the plugin into a new Publish project.

labradon commented:
Great! Thanks a lot for your efforts in re-checking. Merging now...
